snapcraft/_store.py

```diff
@@ -281,3 +283,95 @@ def download(snap_name, channel, download_path, arch):
         raise RuntimeError(
             'Failed to download {} at {} (mismatched SHA)'.format(
                 snap_name, download_path))
+
+
+def _get_snap_data(name, store):
+    search_results = store.cpi.search_package(name, 'stable', None)
+    if not search_results:
+        raise RuntimeError(
+            'Snap {name} cannot be found'.format(
+                name=name))
+    details_url = search_results['_links']['self']['href']
+    details = store.cpi.get(details_url).json()
+    if details.get('result') == 'error':
+        raise RuntimeError(
+            'Error obtaining details for snap {}: {}'.format(
+                name, ', '.join(details['errors'])))
+    return details
+
+
+def gated(snap_name):
+    """Print list of snaps gated by snap_name."""
+    store = storeapi.StoreClient()
+    # Get data for the gating snap
+    with _requires_login():
+        snap_data = _get_snap_data(snap_name, store)
```

Comment:
@ralsina let's drop this search-based query by the new snap-details endpoint caller:

```
store.cpi.get_package(self, snap_name, channel, arch):
````
https://github.com/snapcore/snapcraft/blob/master/snapcraft/storeapi/__init__.py#L291